### PR TITLE
Implements is_valid_version in version chooser to avoid setting a version that does not exists locally

### PR DESCRIPTION
**Pull request by JoaoMario109**

### CODE REVIEW

**Review by joaoantoniocardoso — APPROVED** [review]

File: core/services/versionchooser/test_versionchooser.py [hunk 1]
```diff
@@ -121,11 +123,18 @@ async def test_set_version(write_mock: AsyncMock) -> None:
 async def test_set_version_invalid_settings(json_mock: mock.MagicMock) -> None:
     client = mock.MagicMock()
     chooser = VersionChooser(client)
+
+    # Image will not exists locally, bt for test lets suppose it exists
```
Comment (joaoantoniocardoso) [inline core/services/versionchooser/test_versionchooser.py:127]:
```suggestion
    # Image does not exist locally, but for the test let's fake it exists
```

File: core/services/versionchooser/test_versionchooser.py [hunk 2]
```diff
@@ -216,6 +225,13 @@ async def test_set_version_json_exception(json_mock: mock.MagicMock) -> None:
     client = mock.MagicMock()
     json_mock.side_effect = Exception()
     chooser = VersionChooser(client)
+
+    # Image will not exists locally, bt for test lets suppose it exists
```
Comment (joaoantoniocardoso) [inline core/services/versionchooser/test_versionchooser.py:229]:
```suggestion
    # Image does not exist locally, but for the test let's fake it exists
```

File: core/services/versionchooser/test_versionchooser.py (hunk 1 above)
Comment (JoaoMario109) [reply core/services/versionchooser/test_versionchooser.py]: Added!

File: core/services/versionchooser/test_versionchooser.py (hunk 2 above)
Comment (JoaoMario109) [reply core/services/versionchooser/test_versionchooser.py]: Added!

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**